test/testautovariables.cpp

```diff
@@ -770,6 +772,17 @@ class TestAutoVariables : public TestFixture {
         ASSERT_EQUALS("[test.cpp:4]: (error) Address of auto-variable 'x' returned\n", errout.str());
     }
 
+    // #8615 - Cppcheck claims that address is returned when std::string is assigned a char pointer
+    void returnLocalVariableFalsePositive1() {
+        check( "std::string f() {                  \n"

```

Comment:
please remove the trailing spaces. I wonder if the test case can be reduced. do you need to initialize the vector to get the FP?

Reply:
What's wrong with trailing spaces? I thought that the code would look more clear with them.

Yes, the vector needs to be initialized to get the FP.

Reply:
this stands out. we don't use trailing spaces in the tests.